### PR TITLE
ArcRotateCamera: Allow to pass a transform node to setTarget / targetHost

### PR DESCRIPTION
**Pull request by Popov72**
See https://forum.babylonjs.com/t/update-camera-position-as-an-orbit-around-a-transformnode/46773/3

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14666/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14666/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14666/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14666/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14666/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14666/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14666/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14666/merge#BCU1XR#0

**Review by carolhmj — APPROVED** [review]

File: packages/dev/core/src/Cameras/arcRotateCamera.ts [hunk 1]
```diff
@@ -1078,21 +1079,22 @@ export class ArcRotateCamera extends TargetCamera {
...
      * @param toBoundingCenter In case of a mesh target, defines whether to target the mesh position or its bounding information center
      * @param allowSamePosition If false, prevents reapplying the new computed position if it is identical to the current one (optim)
      * @param cloneAlphaBetaRadius If true, replicate the current setup (alpha, beta, radius) on the new target
      */
-    public setTarget(target: AbstractMesh | Vector3, toBoundingCenter = false, allowSamePosition = false, cloneAlphaBetaRadius = false): void {
+    public setTarget(target: TransformNode | Vector3, toBoundingCenter = false, allowSamePosition = false, cloneAlphaBetaRadius = false): void {
         cloneAlphaBetaRadius = this.overrideCloneAlphaBetaRadius ?? cloneAlphaBetaRadius;
 
-        if ((<any>target).getBoundingInfo) {
+        const canTakeHostPath = (!toBoundingCenter || (toBoundingCenter && (<any>target).getBoundingInfo)) && (target as TransformNode).computeWorldMatrix;
```
Comment (sebavan) [inline packages/dev/core/src/Cameras/arcRotateCamera.ts:1090]: if toBoundingCenter is true and target is a transformNode, I guess we will fall into the else case where target is being used as a Vector3 which may result in a crash ?

File: packages/dev/core/src/Cameras/arcRotateCamera.ts [hunk 2]
```diff
@@ -1087,15 +1087,14 @@ export class ArcRotateCamera extends TargetCamera {
...
         cloneAlphaBetaRadius = this.overrideCloneAlphaBetaRadius ?? cloneAlphaBetaRadius;
 
-        const canTakeHostPath = (!toBoundingCenter || (toBoundingCenter && (<any>target).getBoundingInfo)) && (target as TransformNode).computeWorldMatrix;
-        if (canTakeHostPath) {
-            if (toBoundingCenter) {
+        if ((target as TransformNode).computeWorldMatrix) {
+            if (toBoundingCenter && (<any>target).getBoundingInfo) {
                 this._targetBoundingCenter = (<any>target).getBoundingInfo().boundingBox.centerWorld.clone();
             } else {
                 this._targetBoundingCenter = null;
```
Comment (sebavan) [inline packages/dev/core/src/Cameras/arcRotateCamera.ts:1094]: should this be simply the transformnode position ?

Comment (Popov72) [reply packages/dev/core/src/Cameras/arcRotateCamera.ts:1094]: Yes, the `this._target = this._getTargetPosition();` call on line 1098 will use the transform node position for the target.

**Pull request merged**